src/driver/transport/ssh.cr

```diff
@@ -81,13 +81,39 @@ class PlaceOS::Driver
           # Attempt to authenticate
           supported_methods = session.login_with_noauth(settings.username)
           if supported_methods
-            if password = settings.password
-              session.login(settings.username, password)
-            end
-
-            if prikey = settings.private_key
-              pubkey = settings.public_key.not_nil!
-              session.login_with_data(settings.username, prikey, pubkey, settings.passphrase.try &.to_slice.to_unsafe)
+            if supported_methods.is_a?(Array(String))
+              logger.debug { "supported auhentication methods: #{supported_methods}" }
+
+              supported_methods.each do |auth_method|
+                case auth_method
+                when "publickey"
+                  if prikey = settings.private_key
+                    pubkey = settings.public_key.not_nil!
+                    session.login_with_data(settings.username, prikey, pubkey, settings.passphrase.try &.to_slice.to_unsafe)
```

Comment:
I _think_ that these may raise an exception on unsuccessful auth. May need to catch that within this iteration.

Reply:
Should only error if there is an issue with the session, and then we want it to close anyway

Reply:
Ah ok, that's what I would have been seeing previously (trying to auth with an unsupported auth method, rather than an auth fail).